framework/src/Volo.Abp.Caching/Volo/Abp/Caching/DistributedCache.cs

```diff
@@ -786,12 +1089,33 @@ protected virtual TCacheItem ToCacheItem([CanBeNull] byte[] bytes)
                     )
                 ).ToArray();
         }
-        
+
         private static KeyValuePair<TCacheKey, TCacheItem>[] ToCacheItemsWithDefaultValues(TCacheKey[] keys)
         {
             return keys
                 .Select(key => new KeyValuePair<TCacheKey, TCacheItem>(key, default))
                 .ToArray();
         }
+
+        protected virtual bool ShouldConsiderUow(bool considerUow)
+        {
+            return considerUow && UnitOfWorkManager.Current != null;
+        }
+
+        protected virtual string GetUnitOfWorkCacheKey()
+        {
+            return DistributedCacheName + CacheName;
+        }
+
+        protected virtual Dictionary<TCacheKey, UnitOfWorkCacheItem<TCacheItem>> GetUnitOfWorkCache()
+        {
+            if (UnitOfWorkManager.Current == null)
+            {
+                throw new AbpException($"There is no unit of work in the current context, The {GetType().Name} can only be used in a unit of work.");
```

Comment:
Actually, DistributedCache.cs can be used without UOW. This error message it wrong.
If this method is called out of UOW, ABP Framework has a bug and the error message should indicate that. At least, "There is no active UOW" would be better.